### ISSUE TITLE
On macos after an intense network load, the forwarded ports into the podman container stop responding.

### ISSUE THREAD

**Issue by NTMan** [containers/podman#15095]
/kind bug

**Description**

On macos after an intense network load, the forwarded ports into the podman container stop responding.

**Steps to reproduce the issue:**

1. start container with mapped ports `--network bridge -p 33333:33333 -p 33334:33334 -p 9634:9634 -p 9635:9635 -p 2001:2001 -p 2002:2002 -p 2003:2003 -p 2004:2004`

2. simulate high network activity

3. on the host machine curl comand fail to connect to forwarded ports. For example:
```
% curl "http://localhost:2001/service/?stat"
curl: (28) Failed to connect to localhost port 2001 after 26115 ms: Operation timed out
```
And of course I am checked that this curl command works inside container.

**Output of `podman version`:**

```
% podman --version
podman version 4.1.1
```

**Output of `podman info --debug`:**

```
% podman info --debug
host:
  arch: arm64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 88.45
    systemPercent: 2.51
    userPercent: 9.04
  cpus: 8
  distribution:
    distribution: fedora
    variant: coreos
    version: "36"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
    uidmap:
    - container_id: 0
      host_id: 501
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
  kernel: 5.18.11-200.fc36.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 1396178944
  memTotal: 16729239552
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/501/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.aarch64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 45m 11.06s
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 1
    stopped: 0
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphRootAllocated: 106825756672
  graphRootUsed: 15187017728
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 1
  runRoot: /run/user/501/containers
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 4.1.1
  Built: 1655914710
  BuiltTime: Wed Jun 22 21:18:30 2022
  GitCommit: ""
  GoVersion: go1.18.3
  Os: linux
  OsArch: linux/arm64
  Version: 4.1.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
% brew list podman
/opt/homebrew/Cellar/podman/4.1.1/bin/podman
/opt/homebrew/Cellar/podman/4.1.1/bin/podman-mac-helper
/opt/homebrew/Cellar/podman/4.1.1/bin/podman-remote
/opt/homebrew/Cellar/podman/4.1.1/etc/bash_completion.d/podman
/opt/homebrew/Cellar/podman/4.1.1/libexec/gvproxy
/opt/homebrew/Cellar/podman/4.1.1/share/fish/vendor_completions.d/podman.fish
/opt/homebrew/Cellar/podman/4.1.1/share/man/ (163 files)
/opt/homebrew/Cellar/podman/4.1.1/share/zsh/site-functions/_podman
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical Macbook with M1 Max

**Comment by vrothberg** [comment]
Thanks for reaching out, @NTMan!

@Luap99 PTAL

**Comment by Luap99** [comment]
> simulate high network activity

Can you be more specific?


Any change you can build and test with gvproxy 0.4.0 (https://github.com/containers/gvisor-tap-vsock/releases/tag/v0.4.0)?
Maybe related https://github.com/containers/podman/issues/12495

**Comment by NTMan** [comment]
> Any change you can build and test with gvproxy 0.4.0 (https://github.com/containers/gvisor-tap-vsock/releases/tag/v0.4.0)?

Unfortunately it not helped if I replaced gvproxy right.
1. I downloaded [gvproxy-darwin-arm64](https://github.com/containers/gvisor-tap-vsock/releases/download/v0.4.0/gvproxy-darwin-arm64)
2. I opened directory `/opt/homebrew/Cellar/podman/4.1.1/libexec` and rename original gvproxy to gvproxy.old
3. I copied downloaded gvproxy-darwin-arm64 in place gvproxy and set right chown and chmod.
4. I restarded podman machine and reproduce described issue again.

<img width="1728" alt="Screenshot 2022-08-01 at 04 43 15" src="https://user-images.githubusercontent.com/200750/182050531-e475c610-c99b-456b-a2ed-6bedb098a626.png">

**Comment by binaryfields** [comment]
I have the same issue with latest version of Podman on macOS 13.2.

Client:       Podman Engine
Version:      4.3.1
API Version:  4.3.1
Go Version:   go1.18.8
Built:        Wed Nov  9 15:43:58 2022
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.3.1
API Version:  4.3.1
Go Version:   go1.19.2
Built:        Fri Nov 11 10:00:31 2022
OS/Arch:      linux/arm64

**Comment by binaryfields** [comment]
This issue seems to be fixed by gvproxy 0.5.0. I updated #12495 with details on issues with gvproxy 0.4.0  under heavy network load and the results of testing with 0.5.0.

**Comment by vrothberg** [comment]
Thanks, @binaryfields !  Closing given the issue seems fixed.
